<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>785</width>
    <height>680</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="BasicOpenGLView" name="oglwidget" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: rgb(192, 193, 186);</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QGroupBox" name="groupBox">
        <property name="maximumSize">
         <size>
          <width>125</width>
          <height>590</height>
         </size>
        </property>
        <property name="title">
         <string>Controls</string>
        </property>
        <widget class="QPushButton" name="pushButton">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>30</y>
           <width>114</width>
           <height>32</height>
          </rect>
         </property>
         <property name="text">
          <string>quit</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="translateFieldX">
         <property name="geometry">
          <rect>
           <x>40</x>
           <y>60</y>
           <width>61</width>
           <height>22</height>
          </rect>
         </property>
        </widget>
        <widget class="QLineEdit" name="rotateField">
         <property name="geometry">
          <rect>
           <x>50</x>
           <y>160</y>
           <width>51</width>
           <height>22</height>
          </rect>
         </property>
        </widget>
        <widget class="QPushButton" name="rotateButton">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>180</y>
           <width>110</width>
           <height>32</height>
          </rect>
         </property>
         <property name="text">
          <string>rotate</string>
         </property>
        </widget>
        <widget class="QPushButton" name="undoButton">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>560</y>
           <width>110</width>
           <height>32</height>
          </rect>
         </property>
         <property name="text">
          <string>Pop Matrix</string>
         </property>
        </widget>
        <widget class="QPushButton" name="translateButton">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>110</y>
           <width>110</width>
           <height>32</height>
          </rect>
         </property>
         <property name="text">
          <string>translate</string>
         </property>
        </widget>
        <widget class="QPushButton" name="scaleButton">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>270</y>
           <width>110</width>
           <height>32</height>
          </rect>
         </property>
         <property name="text">
          <string>scale</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="scaleFieldX">
         <property name="geometry">
          <rect>
           <x>40</x>
           <y>220</y>
           <width>61</width>
           <height>22</height>
          </rect>
         </property>
        </widget>
        <widget class="QLineEdit" name="translateFieldY">
         <property name="geometry">
          <rect>
           <x>40</x>
           <y>90</y>
           <width>61</width>
           <height>22</height>
          </rect>
         </property>
        </widget>
        <widget class="QLineEdit" name="scaleFieldY">
         <property name="geometry">
          <rect>
           <x>40</x>
           <y>250</y>
           <width>61</width>
           <height>22</height>
          </rect>
         </property>
        </widget>
        <widget class="QLabel" name="label">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>60</y>
           <width>31</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>x:</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_2">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>220</y>
           <width>31</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>x:</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_3">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>250</y>
           <width>31</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>y:</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_4">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>90</y>
           <width>31</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>y:</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_5">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>160</y>
           <width>51</width>
           <height>20</height>
          </rect>
         </property>
         <property name="text">
          <string>angle:</string>
         </property>
        </widget>
        <widget class="QPushButton" name="undoButton_2">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>400</y>
           <width>110</width>
           <height>32</height>
          </rect>
         </property>
         <property name="text">
          <string>Push Matrix</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="mField1">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>310</y>
           <width>31</width>
           <height>22</height>
          </rect>
         </property>
        </widget>
        <widget class="QLineEdit" name="mField2">
         <property name="geometry">
          <rect>
           <x>50</x>
           <y>310</y>
           <width>31</width>
           <height>22</height>
          </rect>
         </property>
        </widget>
        <widget class="QLineEdit" name="mField3">
         <property name="geometry">
          <rect>
           <x>90</x>
           <y>310</y>
           <width>31</width>
           <height>22</height>
          </rect>
         </property>
        </widget>
        <widget class="QLineEdit" name="mField6">
         <property name="geometry">
          <rect>
           <x>90</x>
           <y>340</y>
           <width>31</width>
           <height>22</height>
          </rect>
         </property>
        </widget>
        <widget class="QLineEdit" name="mField5">
         <property name="geometry">
          <rect>
           <x>50</x>
           <y>340</y>
           <width>31</width>
           <height>22</height>
          </rect>
         </property>
        </widget>
        <widget class="QLineEdit" name="mField4">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>340</y>
           <width>31</width>
           <height>22</height>
          </rect>
         </property>
        </widget>
        <widget class="QLineEdit" name="mField9">
         <property name="geometry">
          <rect>
           <x>90</x>
           <y>370</y>
           <width>31</width>
           <height>22</height>
          </rect>
         </property>
        </widget>
        <widget class="QLineEdit" name="mField8">
         <property name="geometry">
          <rect>
           <x>50</x>
           <y>370</y>
           <width>31</width>
           <height>22</height>
          </rect>
         </property>
        </widget>
        <widget class="QLineEdit" name="mField7">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>370</y>
           <width>31</width>
           <height>22</height>
          </rect>
         </property>
        </widget>
        <widget class="QTextEdit" name="currentStack">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>440</y>
           <width>111</width>
           <height>101</height>
          </rect>
         </property>
        </widget>
        <widget class="QLabel" name="label_6">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>540</y>
           <width>62</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>Stack</string>
         </property>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>785</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>BasicOpenGLView</class>
   <extends>QWidget</extends>
   <header location="global">basicopenglview.h</header>
   <container>1</container>
   <slots>
    <slot>rotate()</slot>
    <slot>transform()</slot>
    <slot>translate()</slot>
    <slot>undo()</slot>
    <slot>scale()</slot>
    <slot>applyMatrix()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>631</x>
     <y>106</y>
    </hint>
    <hint type="destinationlabel">
     <x>670</x>
     <y>49</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>undoButton</sender>
   <signal>clicked()</signal>
   <receiver>oglwidget</receiver>
   <slot>undo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>633</x>
     <y>569</y>
    </hint>
    <hint type="destinationlabel">
     <x>568</x>
     <y>478</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>translateButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>translate()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>688</x>
     <y>174</y>
    </hint>
    <hint type="destinationlabel">
     <x>734</x>
     <y>180</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rotateButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>rotate()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>695</x>
     <y>247</y>
    </hint>
    <hint type="destinationlabel">
     <x>733</x>
     <y>246</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>scaleButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>scale()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>691</x>
     <y>336</y>
    </hint>
    <hint type="destinationlabel">
     <x>732</x>
     <y>337</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>undoButton_2</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>applyMatrix()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>687</x>
     <y>505</y>
    </hint>
    <hint type="destinationlabel">
     <x>733</x>
     <y>505</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>transform()</slot>
  <slot>animate()</slot>
  <slot>translate()</slot>
  <slot>rotate()</slot>
  <slot>scale()</slot>
  <slot>applyMatrix()</slot>
 </slots>
</ui>
